packages/ccip-svm/src/__tests__/evmExtraArgsV2.test.ts

```diff
@@ -0,0 +1,179 @@
+
+            expect(consoleSpy).not.toHaveBeenCalled()
+            expect(encoded).toBeInstanceOf(Uint8Array)
+            expect(encoded.length).toBe(68)
+            expect(encoded).toEqual(new Uint8Array(Buffer.from(expected.slice(2), 'hex')))
+        })
+
+        it('should include correct tag at the beginning', () => {
+            const encoded = encodeEVMExtraArgsV2({ gasLimit: 0n })
+
+            // Extract first 4 bytes and convert to uint32 big-endian
+            const tag = new DataView(encoded.buffer, encoded.byteOffset, 4).getUint32(0, false)
+            expect(tag).toBe(EVM_EXTRA_ARGS_V2_TAG)
+            expect(tag).toBe(0x181dcf10)
+        })
+    })
+
+    describe('decodeEVMExtraArgsV2', () => {
+        it('should decode EVM extra args correctly with default values', () => {
+            const original = {
+                gasLimit: 200000n,
+                allowOutOfOrderExecution: true
+            }
+
+            const encoded = encodeEVMExtraArgsV2(original)
+            const decoded = decodeEVMExtraArgsV2(encoded)
+
+            expect(decoded.gasLimit).toBe(original.gasLimit)
+            expect(decoded.allowOutOfOrderExecution).toBe(original.allowOutOfOrderExecution)
+        })
+
+        it('should decode EVM extra args with allowOutOfOrderExecution=false', () => {
+            const original = {
+                gasLimit: 150000n,
+                allowOutOfOrderExecution: false
+            }
+
+            const encoded = encodeEVMExtraArgsV2(original)
+            const decoded = decodeEVMExtraArgsV2(encoded)
+
+            expect(decoded.gasLimit).toBe(original.gasLimit)
+            expect(decoded.allowOutOfOrderExecution).toBe(original.allowOutOfOrderExecution)
+        })
+
+        it('should decode token-only transfers (gasLimit = 0)', () => {
+            const original = {
+                gasLimit: 0n,
+                allowOutOfOrderExecution: true
+            }
+
+            const encoded = encodeEVMExtraArgsV2(original)
+            const decoded = decodeEVMExtraArgsV2(encoded)
+
+            expect(decoded.gasLimit).toBe(0n)
+            expect(decoded.allowOutOfOrderExecution).toBe(true)
+        })
+
+        it('should throw error for data too short', () => {
+            const shortData = new Uint8Array(10) // Too short
+
+            expect(() => decodeEVMExtraArgsV2(shortData)).toThrow('Invalid EVM Extra Args V2: data too short')
+        })
+
+        it('should throw error for invalid tag', () => {
+            const invalidData = new Uint8Array(68)
+            // Set wrong tag (first 4 bytes)
+            invalidData[0] = 0x12
+            invalidData[1] = 0x34
+            invalidData[2] = 0x56
+            invalidData[3] = 0x78
+
+            expect(() => decodeEVMExtraArgsV2(invalidData)).toThrow(
+                'Invalid EVM Extra Args V2 tag: expected 0x181dcf10, got 0x12345678'
+            )
+        })
+    })
+
+    describe('round-trip encoding/decoding', () => {
```

Comment:
Can we not just have this one test and get rid of the individual it blocks? this test table exercises the same and is more compact/easier to maintain?